### PR TITLE
feat: [IAC-650]: Add iacm terraform plugin (Hotfix)

### PR DESCRIPTION
**Pull request by bgribben-harness**
Hotfixes the changes introduced in https://github.com/harness/harness-core-ui/pull/15950 to 0.349.x

### CODE REVIEW

File: src/modules/85-iacm/components/IACMPipelineSteps/IACMTerraformPlugin/index.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  protected defaultValues = {
+    type: StepType.IACMTerraformPlugin,
+    name: '',
+    identifier: '',
+    spec: {
+      command: ''
+    }
+  }
+
+  /* istanbul ignore next */
```
Comment (knagurski) [inline src/modules/85-iacm/components/IACMPipelineSteps/IACMTerraformPlugin/index.tsx:50]: Very naughty! Just ignoring all the things is a bit sus

File: src/modules/85-iacm/components/IACMPipelineSteps/IACMTerraformPlugin/__tests__/IACMTerraformPluginVariableStep.test.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+const renderComponent = (props: any) => {
+  return render(
+    <TestWrapper>
+      <IACMTerraformPluginVariableStep
+        initialValues={props.initialValues}
+        {...{
+          stageIdentifier: 'qaStage',
+          metadataMap: props.metadataMap,
+          variablesData: props.variablesData
+        }}
```
Comment (knagurski) [inline src/modules/85-iacm/components/IACMPipelineSteps/IACMTerraformPlugin/__tests__/IACMTerraformPluginVariableStep.test.tsx:27]: This is just kinda weird.
```suggestion
        stageIdentifier="qaStage"
        metadataMap={props.metadataMap}
        variablesData={props.variablesData}
```

**Review by knagurski — APPROVED** [review]
Looks good for a hotfix

**Review by knagurski — APPROVED** [review]
Looks good for a hotfix

**Review by udayvunnam — APPROVED** [review]

**Pull request merged**